Changes for SN v1.1.0 (#1051)
hashicorp/terraform-website: content/source/docs/cloud/integrations/service-now/index.html.md

--|--
`tf_var_VARIABLE_NAME` | Terraform Variable: `VARIABLE_NAME`
`tf_env_ENV_NAME` | Environment Variable: `ENV_NAME`

> `sensitive_tf_var_VARIABLE_NAME` | Sensitive Terraform Variable (Write Only): `VARIABLE_NAME`

Reviewer: Are you considering adding an equivalent setting for sensitive environment variables? I think those might be more widely used, since most providers use env vars for their credentials.

For users who only need to order from the Terraform Catalog, we recommend creating another role with read-only permissions for `x_terraform_vcs_repositories` to view the available repositories for ordering infrastructure.

> ## Example Customizations

Reviewer: This needs some reorganization, beyond just the prose style fixes. 

- It's extremely long, on an already overlong mono-page. This should be moved out into a new page, and the ServiceNow integration item in the sidebar should be turned into a section. While doing this, it's worth checking whether other things should move; note that this page bills itself as "setup instructions," so usage information probably belongs in a separate page. 
- The header levels are somewhat messed up -- we go directly from H3 to H5. Treat headers as indicators of the document's outline level, and only jump down one level at a time. 
- The navigation instructions are quite complex, and a couple of screenshots (1024px wide) would go a long way to help.

### Creating a Catalog Item with Pinned Variables

> This example use case is used to demo a potential solution to creating a Terraform Catalog Item for resources that limits user input to certain variables.

Reviewer: ```suggestion
This example use case creates a Terraform catalog item for resources that limits user input to certain variables.
```

I'm seeing a few instances through here that look like they shouldn't be capitalized. Let me know if these are brand names, or should be a proper noun for some other reason!

##### Create Service Catalog Item

1. Enter the ServiceNow Studio
1. Click "Create Application File"
    - Select "Catalog Item" under "Service Catalog" > Click "Create"

> 1. Name the new Catalog Item. (Our example will be `Example With Pinned Variables`)

Reviewer: ```suggestion
1. Name the new catalog item. (The rest of this example assumes an item named `Example With Pinned Variables`.)
```

1. Select Catalogs: "Terraform Catalog" > Select Categories: "Terraform Resources"
1. Add any other descriptions you may want.
1. Click "Submit"

##### Create Variable Set

1. Click "Create Application File"
1. Select "Variable Set" > "Service Catalog" > Click "Create"

> 1. Name your variable set (Our example will be `Example Pinned Variables`)

Reviewer: ```suggestion
1. Name your variable set. (The rest of this example assumes a set named `Example Pinned Variables`, with an "Internal Name" of `example_pinned_variables`.)
```

>     - This will default "Internal Name" to `example_pinned_variables`

Reviewer: ```suggestion
```

1. Click "Submit"

> 1. Under the "Variables" tab click "New"

Reviewer: ```suggestion
1. Under the "Variables" tab, click "New".
```

>     1. Create your variable:

Reviewer: ```suggestion
```

>     1. In this example we will create a field called `tf_var_pet_name_length` that will be for a Terraform variable that determines the number of words to use for the pet server name.

Reviewer: ```suggestion
    1. Create your variable by providing "Question" and "Name" values. For example, to create a field called `tf_var_pet_name_length` to limit the length of a pet server name:
```

        - Question: `Pet Name Length`
        - Name: `tf_var_pet_name_length`

>         - `tf_var_` tells the Terraform ServiceNow SDK that this is a Terraform Variable.

Reviewer: ```suggestion
        - `tf_var_` tells the Terraform ServiceNow SDK that this is a Terraform variable.
```

    1. Repeat variable creation as necessary for your use case.

>     1. Click "Submit"

Reviewer: ```suggestion
    1. Click "Submit".
```

##### Add Variable Set to Catalog Item

> 1. Go back to your Catalog Item

Reviewer: ```suggestion
1. Go back to your catalog item.
```

> 1. Select "Variable Sets Tab" > Click "New"

Reviewer: ```suggestion
1. Under the "Variable Sets" tab,  click "New".
```

> 1. Search for the Variable Set that was created (`Example Pinned Variables`)

Reviewer: ```suggestion
1. Search for the variable set that was created (`Example Pinned Variables`).
```

> 1. Click "Submit"

Reviewer: ```suggestion
1. Click "Submit".
```

##### Create Custom Workflow

We will use the "Example With Pinned Variables" example workflow for our custom workflow.

In the ServiceNow Studio:

1. Open the "Example With Pinned Variables", Workflow > Workflow > "Example With Pinned Variables"
1. Double click on the "Run Script" item
1. Copy the script code from the editor
1. Close the "Run Script" and "Workflow" window
1. Create a new custom workflow: Create Application File > Workflow
    - Name: `Example Pinned Variables` (replace with your custom workflow name)
    - Table: `Requested Item [sc_req_item]` and Click "Submit"
1. Add a "Run Script" item. Core Tab > Utilities > Run Script
1. Name: `Example Workflow Pinned Variables`
1. Script:

>     - Paste in the code copied from "Provision Resources with Variables"

Reviewer: This is the first time this is mentioned in this process. Did you miss adding something above, or mis-copy a name?

Author: It was mentioned a few steps above. I made an edit to hopefully clarify it better.

    - Change the following lines:

```javascript
  // ** Custom Variables **
  var VAR_SET_NAME = "example_pinned_variables";
  var VCS_REPO     = "ORG_NAME/REPO_NAME";
```

> 1. Click Submit

Reviewer: ```suggestion
1. Click "Submit"
```
